templates/cluster-template-aad.yaml

```diff
@@ -210,13 +210,15 @@ spec:
 apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
 kind: AzureClusterIdentity
 metadata:
-  name: ${CLUSTER_IDENTITY_NAME}
+  labels:
+    clusterctl.cluster.x-k8s.io/move-hierarchy: "true"
+  name: ${CLUSTER_IDENTITY_NAME:=cluster-identity}
   namespace: default
 spec:
   allowedNamespaces: {}
   clientID: ${AZURE_CLIENT_ID}
   clientSecret:
-    name: ${AZURE_CLUSTER_IDENTITY_SECRET_NAME}
-    namespace: ${AZURE_CLUSTER_IDENTITY_SECRET_NAMESPACE}
+    name: ${AZURE_CLUSTER_IDENTITY_SECRET_NAME:=cluster-identity-secret}
```

Comment:
Why all the added defaults in this PR? I recall we decided to leave those without defaults so it would be a reminder to folks to create the secret when using the new templates

Reply:
oh ok, I'll remove them, thought it would be useful, but good point